MapboxNavigation/RouteMapViewController.swift

```diff
@@ -564,6 +564,10 @@ extension RouteMapViewController: RoutePageViewControllerDelegate {
     }
 
     func stepBefore(_ step: RouteStep) -> RouteStep? {
+        guard let index = routeController.routeProgress.currentLeg.steps.index(of: step),
```

Comment:
Nit: this could also be written `routeController.routeProgress.currentLegProgress.leg.steps.index(of: step)`, which would enable us to factor out `routeController.routeProgress.currentLegProgress` into a local variable. It wouldn’t be a significant performance optimization, just a way to keep things short and sweet.